### DISCUSSION TITLE
Issues running CJBCA-CE on Synology (DSM7)

### DISCUSSION THREAD

**Discussion by t984447** [Keyfactor/ejbca-ce#152] [Q&A] [answered]
Hello,

Ive been trying now for a while to setup CJBCA-CE Docker container on Synology (DSM 7.1.1-42962 Update 1)
and am having issues getting it to store data on shared volumes.
The docker container does work if I leave the settings be but this also means that all settings will be lost upon updating/replacing the container.

When mounting a volume to one of the shared folders in the NAS I need to make sure that the user running in the container har permission to access those volumes. But I cannot find any information in the documents on how to pre-define which users to use or how to find the selected user.

Has anyone found a way to curb this issue?

**Comment by primetomas** [comment, marked as answer]
What do you want to do with shared volumes?

EJBCA does not store data as such on the file system and each EJBCA container running should be independent of each other, except for the data in the database. To persist or cluster installations you should simply use an external database. See https://hub.docker.com/r/keyfactor/ejbca-ce

**Reply by t984447** [reply]
I might have missunderstod the way its meant to work. 

When a new version of the container i released the update will replaced the whole container, and thus all settings/certiciates will be reset.
The idea was then to have configurations and logs placed on a volume which the containers always could be refered to.

It did not occur to me that everything might be stored in the DB for presistance between upgrades.


Thank you for the response, I will try installing a DB server on the NAS for Ejbca.

**Reply by primetomas** [reply]
Correct, with a persistent DB, the container itself should be ephemeral.
